layer/support/trace_event_logging.cc

```diff
@@ -33,6 +33,13 @@ constexpr size_t kMaxBufferSize = 128;
 
 std::string DoubleToString(double value) {
   std::string str;
+#if defined(__ANDROID__)
+  // stdlib on Android doesn't seem to support to_chars although the c++17
+  // flag should allow the compiler to use it (c++17 ops was removed).
+  std::ostringstream strs;
+  strs << value;
+  str = strs.str();
+#else
```

Comment:
IIRC we used to use stringstream but were run into issues with formats. Here, we'd want the format to be locale independent. Is there something else we could use instead like `sprintfn`?

I'd prefer if we used the same implementation for all targets. If that's not possible, could you move this alternative formatting to a new function and only guard the function call with ifdefs?

Reply:
The property I care about the most is being able to produce a trace long on one device and read it on another device, getting the same results on both ends. Not sure what the best way to handle floats is -- if a simple ostringstream unblocks you, this is fine with me as long as there's a comment explaining the reasoning behind it and/or a TODO for platform-agnostic float printing.

Reply:
OK, so I switched to a very simple custom implementation that is platform-agnostic and locale-independent. It should be in the format that most humans would use. It generates redundant zeros (e.g. 123.456000) in a fully predictable manner, so I updated the tests cases to reflect that.
I also wrote a note that to_chars should be used once Android supports it.

Reply:
Perfect, thank you!